woocommerce/woocommerce-ios
# [Mobile Payments] Model objects for CardReader and PaymentIntent

Closes #3649 

⚠️  This PR is not against develop, but against the future branch for Mobile Payments ⚠️ 

We continue building our prototype. As a follow up to #3643, we need some model objects. 
For reference: p91TBi-4q4-p2

Scope:
* Enough code to implement CardReader, PaymentIntent, and Charge. These objects model some of their counterparts in the Stripe Terminal SDK. For more info and APIs: https://stripe.dev/stripe-terminal-ios/docs/Classes/SCPReader.html
Initializers to instantiate our domain objects with their counterparts in StripeTerminal.

Out of scope:
* Unit tests. We can not test our model objects directly, because for that we would need to create mock objects on the StripeTerminal side of the boundary. However, the initializers of the model objects in StripeTerminal are marked as NS_UNAVAILABLE. We can not even instantiate subclasses.
* Unit tests will come in a later PR, in the form of integration tests. We will use the test mode of the SDK to simulate calls to the methods that return these model objects which we will then proceed to unit test.

## Changes
* Added model objects for CardReader, PaymentIntent, Charge and enums to back up the different statuses (i.e. PaymentIntentStatus, ChargeStatus...)

## How to test
* Look at the code. 
* This code is not executed for now, so this PR does not add any new behaviour
* CI should be ✅ 

Update release notes:
